game_api.txt

```diff
@@ -83,6 +83,16 @@ in `bones.player_inventory_lists`.
 
 e.g. `table.insert(bones.player_inventory_lists, "backpack")`
 
+Additionally, callbacks can be registered to transfer items into the bones on death:
+
+`bones.public.register_transfer_inventory_to_bones_on_player_death(function(player){})`
+
+In the above functions this routine should be used to add itmes to the bones inventory:
+
+`bones.public.transfer_stack_to_bones(stack)`
```

Comment:
Same here. I wonder whether the slightly "functional" returning a list to be added might be cleaner as opposed to this "imperative" style, especially considering the ugly global context in which this function operates: What happens if this is called when no bones are currently being built?

Get rid of this function in its current form, either by providing the callback with a closure parameter (`function(name, add_item_to_bones)`) such that it can then call its local `add_item_to_bones` to well, add items to bones. Or return a list.

Reply:
> What happens if this is called when no bones are currently being built?
it would crash the whole server due to indexing a nil. That was one of the reasons it was done this way: I wanted to discourage everyone from using this routine outside of this context (however as per other requests the context will now be stored in a function local variable and passed as function argument)